File: app/src/main/java/com/inqint/yarnrequirements/Projects/ShortLengthUnits.java
package com.inqint.yarnrequirements.Projects;

/* Enumeration for shorter units: inches or cm */
public enum ShortLengthUnits { inches, cm;
    private static ShortLengthUnits[] values = null;
    public static ShortLengthUnits fromInt(int i) {
        if (ShortLengthUnits.values == null) {
            ShortLengthUnits.values = ShortLengthUnits.values();
        }
        return ShortLengthUnits.values[i];
    }
}

File: app/src/main/java/com/inqint/yarnrequirements/Projects/ShortLengthUnits.kt
package com.inqint.yarnrequirements.Projects
import java.io.Serializable


/* Enumeration for shorter units: inches or cm */
enum class ShortLengthUnits : Serializable {
    inches, cm;


    companion object {
        private var values: Array<ShortLengthUnits>? = null
        fun fromInt(i: Int): ShortLengthUnits {
            if (ShortLengthUnits.values == null) {
                ShortLengthUnits.values = ShortLengthUnits.values()
            }
            return ShortLengthUnits.values!![i]
        }
    }
}
